nsapass

```diff
@@ -456,32 +434,31 @@ def chpass():
 # arguments are specified by their named arguments to `add_argument`.
 # 'PASSWORD' stands for `[-g LEN SET | -m]` and is handled separately.
 allcmds = [
-    ['get', ['username','index','-a','-z'],
+    ['get', ['id','rev','-o','-z'],
      {'help':'load a password to clipboard'}],
-    ['add', ['-a','-z','PASSWORD','-u','-i','-n'],
+    ['add', ['id','-z','PASSWORD','-n'],
      {'help':'add an entry'}],
-    ['del', ['username','index','-z'],
+    ['del', ['id','rev','-z'],
      {'help':'delete an entry'}],
-    ['undel', ['username','index','-z'],
+    ['undel', ['id','rev','-z'],
      {'help':'delete an entry'}],
-    ['mod', ['username','index','-z','PASSWORD','-u','-i','-n'],
+    ['mod', ['id','rev','-z','PASSWORD','-u','-n'],
      {'help':'modify an entry'}],
-    ['view', ['-a','-z'],
+    ['view', ['-o','-z'],
      {'help':'view entries'}],
     ['chpass', [],
      {'help':"change databases's password"}]
 ]
 allargs = {
-    'username':
-        {'metavar':'USERNAME', 'type':str, 'nargs':'?',
-            'help':'query a username'},
-    'index':
-        {'metavar':'INDEX', 'type':int, 'nargs':'?',
+    'id':
+        {'metavar':'ID', 'type':str, 'nargs':'?',
+            'help':'query a id'},
+    'rev':
+        {'metavar':'REV', 'type':int, 'nargs':'?',
             'help':"query by entry's position"},
-    '-a': {'action':'store_true', 'help':'work on the archive'},
+    '-o': {'action':'store_true', 'help':'work on old revisions'},
```

Comment:
i'm not sure if the 2 are mutually exclusive.  e.g.:

- old revisions of (a given entry) are usable for offering a fool-proof system by being able to undo changes (since passwords are critical and better not risk it to accidental changes).
- archived entries become treated as if deleted (and get accessed after `-a` is used).

i think this is mostly my fault for pushing a confusing feature/design at 1st.  i think now you're trying to shake my previous bad design somehow to make sense out of it 😄

Reply:
One big change is that there are no archived entries anymore, only old revisions.

Reply:
yes, the miniality is good.  worth thinking about.  i'll re-think about it.